test-framework/core/src/main/java/org/keycloak/testframework/server/EmbeddedKeycloakServer.java

```diff
@@ -18,6 +23,27 @@ public void start(KeycloakServerConfigBuilder keycloakServerConfigBuilder) {
             builder.addDependency(dependency.getGroupId(), dependency.getArtifactId(), "");
         }
 
+        Set<Path> configFiles = keycloakServerConfigBuilder.toConfigFiles();
+        if (!configFiles.isEmpty()) {
+            Path homeDir = Platform.getPlatform().getTmpDirectory().toPath();
+            Path conf = homeDir.resolve("conf");
+
+            if (!conf.toFile().exists()) {
+                conf.toFile().mkdirs();
+            }
+
+            for (Path configFile : keycloakServerConfigBuilder.toConfigFiles()) {
+                try {
+                    Files.copy(configFile, conf.resolve(configFile.getFileName()));
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+
+            builder.setHomeDir(homeDir);
```

Comment:
@mhajas Do you think that there can be another chance to use this homedir option? My point is, if yes, maybe it would be better to extract it from this if statement and set it by default.

Reply:
I am not sure there will be another usage for homeDir but I extracted it to be a class field so it is more visible we set it.